.github/workflows/update-js-deps.yml

```diff
@@ -0,0 +1,139 @@
+name: js-upgrade
+
+on:
+  workflow_call:
+
+jobs:
+  context:
+    name: Context
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Context
+        run: |
+          # https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability
+          echo "github.base_ref: ${{ github.base_ref }}"
+          echo "github.event_name: ${{ github.event_name }}"
+          echo "github.head_ref: ${{ github.head_ref }}"
+          echo "github.ref: ${{ github.ref }}"
+          echo "github.ref_name: ${{ github.ref_name }}"
+          echo "github.ref_type: ${{ github.ref_type }}"
+          echo "github.repository: ${{ github.repository }}"
+          echo "github.sha: ${{ github.sha }}"
+          echo "job.status: ${{ job.status }}"
+          # https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows
+          echo "github.event.issue.pull_request: ${{ github.event.issue.pull_request }}"
+          echo "github.event.issue.number: ${{ github.event.issue.number }}"
+          echo "github.event.repository.name: ${{ github.event.repository.name }}"
+
+  js-upgrade:
+    name: 'JS upgrade'
+    runs-on: ubuntu-latest
+    env:
+      pr_title: DEP Update JS dependencies
+      pr_desc: Automated yarn upgrade and yarn build
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Read .nvmrc
+        run: echo ::set-output name=version::$(cat .nvmrc)
+        id: read-nvm
+
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ steps.read-nvm.outputs.version }}
+
+      - name: Install yarn
+        run: npm install --global yarn
+
+      - name: Install admin JS
+        if: github.event.repository.name != 'silverstripe-admin'
+        run: |
+          # Install admin js in sibling directory so shared components are available
+          DIR=$(pwd)
+          cd ..
+          git clone https://github.com/silverstripe/silverstripe-admin.git
+          mv silverstripe-admin admin
+          cd admin
+          git checkout 1
+          yarn install
+          cd $DIR
```

Comment:
Can't we just run a composer install here?

Reply:
maybe, however:
- we'd install a bunch of other stuff we don't need, so it's slower
- admin would end up in the `./vendor/silverstripe/admin` directory, rather than the `./../admin` sibling directory, no even sure if the js importer would find the admin js

Reply:
All [our packages are set up](https://github.com/silverstripe/silverstripe-asset-admin/blob/546d37ee78bd6d605a4770e9e3c5839c274201be/package.json#L36-L41) so they can find their dependencies in a root vendor folder as well. That's how all our current build work.

The original build system is designed with the assumption that some packages could depend on other module aside from `silverstripe/admin` ... so it would be safer in the long run to use composer admin rather than manually fetch `siverstripe/admin`.

Reply:
:thinking: even if we did start using another module an a central component repositiory, which honestly seems very unlikely at this stage, then we'd still need to update this script to manually `cd somemodule && yarn install && cd $DIR`

So I don't think we're really achieving by running composer install here other than installing somethings we don't need